src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -45,34 +60,36 @@ public List<Station> getStations() {
             return Collections.emptyList();
         }
 
-        Station upStationTerminal = getUpStationTerminal();
-        Station downStationTerminal = getDownStationTerminal();
+        List<Station> sortedStations = new ArrayList<>();
 
-        List<Station> stations = new ArrayList<>();
+        Station station = getUpStationTerminal();
+        sortedStations.add(station);
 
-        Station upStation = upStationTerminal;
-        while (true) {
-            Section section = findSectionByUpStation(upStation);
-            stations.add(section.getUpStation());
-            if (downStationTerminal.equals(section.getDownStation())) {
-                stations.add(section.getDownStation());
-                break;
-            }
-            upStation = section.getDownStation();
+        while (!station.equals(getDownStationTerminal())) {
+            Section section = findSectionHasUpStation(station);
+            station = section.getDownStation();
+            sortedStations.add(station);
         }
-        return stations;
+
+        return sortedStations;
+    }
+
+    private long findNextSectionCount(Station station) {
+        return sections.stream()
+                .filter(it -> it.getUpStation().equals(station))
+                .count();
```

Comment:
집계된 count를 활용하진 않으니 boolean 리턴해도 좋을 것 같아요 
```suggestion
    private boolean findNextSectionCount(Station station) {
        return sections.stream()
                .anyMatch(it -> it.isSameUpStation(station));
```